mycore-restapi/src/main/java/org/mycore/restapi/v1/utils/MCRRestAPIUploadHelper.java

```diff
@@ -154,27 +160,55 @@ public static Response uploadDerivate(UriInfo info, HttpServletRequest request,
         try {
             MCRObject mcrObj = MCRMetadataManager.retrieveMCRObject(mcrObjIDObj);
             MCRObjectID derID = null;
-            if (overwriteOnExistingLabel) {
-                for (MCRMetaLinkID derLink : mcrObj.getStructure().getDerivates()) {
-                    if (label.equals(derLink.getXLinkLabel()) || label.equals(derLink.getXLinkTitle())) {
-                        derID = derLink.getXLinkHrefID();
+            if (overwriteOnExisting) {
```

Comment:
This Snipped overwrites the last Derivate with the same label, but if a derivate exists with the same classification then it will overwrite it instead of the label one. This seems a little bit confusing.

Reply:
This is because label is deprecated and can be cleaned up for the next release. Later we should also check how restapi v2 handles such things.